doc/source/data/accessing-datasets.rst

```diff
@@ -4,47 +4,188 @@
 Accessing Datasets
 ===================
 
-Datasets can be passed to Ray tasks or actors and accessed with ``.iter_batches()`` or ``.iter_rows()``.
+The data underlying a ``Dataset`` can be accessed in several ways:
+
+* Retrieving a limited prefix of rows.
+* Iterating over rows and batches.
+* Converting into a Torch dataset or a TensorFlow dataset.
+* Converting into a RandomAccessDataset for random access.
+
+Retrieving limited set of rows
+==============================
+
+A limited set of rows can be retried from a ``Dataset`` via the
+:meth:`ds.take() <ray.data.Dataset.take>` API, along with its sibling helper APIs
+:meth:`ds.take_all() <ray.data.Dataset.take_all>`, for retrieving **all** rows, and
```

Comment:
100k by default actually? We should probably warn the risk of actually taking all rows to a single node.

Reply:
There's a warning in the code and we emphasize that it will consume all rows, so I think that suffices. Lmk if you think there should be an explicit warning block, we can do that as a follow-up